src/index.ts

```diff
@@ -746,6 +746,25 @@ const handleFilterOperator = <
 			filterStr = `${operator.slice(1)}(${alias}:${filterStr})`;
 			return addParentKey(filterStr, parentKey, '/');
 		}
+		case '$fn': {
+			const filterx = filter as FilterType<typeof operator, T>;
+			const scope = filterx.$scope;
+			const method = filterx.$method;
+			const args = (filterx.$args ?? []).map((v) => `${escapeValue(v)}`);
+			if (scope == null) {
+				throw new Error(
+					`Function expression (${operator}) has no scope defined.`,
+				);
+			}
+			if (method == null) {
+				throw new Error(
+					`Function expression (${operator}) has no method defined.`,
+				);
+			}
+
+			const filterStr = `${scope}.${method}(` + args.join(',') + ')';
```

Comment:
These may need `escapeResource` called on them potentially? Look around and see if that matches what is done elsewhere (basically if the scope was `$MyScope` then the `$` should probably be escaped as `%??`)

Reply:
Yep, this makes sense. Done. And added a test for it

Reply:
Any reason you didn't user template strings all the way v ?
```suggestion
			const filterStr = `${scope}.${method}(${args.join(',')})`;
```
Balena-lint v9 will be happier I would expect :)